tests/unit/test_task.py

```diff
@@ -153,5 +154,23 @@ def test___close_on_exit___leak_task___raises_resource_warning(interpreter: Mock
     task = Task("MyTask")
 
     with pytest.warns(ResourceWarning):
-        del task
-        gc.collect()
+        task.__del__()
+
+    task.close()
+
+
+def test___close_on_exit_with_grpc_options___leak_task___resource_warning_not_raised(
+    interpreter: Mock, mocker: MockerFixture
+):
+    expect_create_task(interpreter, "GrpcTaskHandle")
+    expect_get_task_name(interpreter, "GrpcTask")
+    grpc_options = create_grpc_options(mocker)
+
+    task = Task("GrpcTask", grpc_options=grpc_options)
+
+    with warnings.catch_warnings(record=True) as warnings_raised:
+        task.__del__()
+    
+    assert len(warnings_raised) == 0
+
+    task.close()
```

Comment:
The test function should have a max of 3 sections (arrange, act, assert), separated by blank lines. This one has 5.

```suggestion
    expect_create_task(interpreter, "GrpcTaskHandle")
    expect_get_task_name(interpreter, "GrpcTask")
    grpc_options = create_grpc_options(mocker)
    task = Task("GrpcTask", grpc_options=grpc_options)

    with warnings.catch_warnings(record=True) as warnings_raised:
        task.__del__()
    
    assert len(warnings_raised) == 0
    task.close()
```

Reply:
Applied the suggestion.